##### To filter the cash flow of each stock for the record closest to the IPO date.

In [1]:
import pandas as pd
import datetime
from datetime import timedelta

In [2]:
d_parser = lambda x: pd.datetime.strptime(x,'%Y-%m-%d')
IPO_df = pd.read_csv('raw_dataset/IPO_Listing_2019_2020.csv', parse_dates = ['IPO_DATE'], date_parser = d_parser)
IPO_df

C:\Users\User\anaconda3\envs\PythonData\lib\site-packages\ipykernel_launcher.py:1: FutureWarning: The pandas.datetime class is deprecated and will be removed from pandas in a future version. Import from datetime module instead.
  """Entry point for launching an IPython kernel.


IPO_DATE SYMBOL                             NAME  IPO_PRICE  CURRENT  \
0   2020-12-29    MRM               Medirom Healthcare       15.0    10.30   
1   2020-12-24   VTAQ          Ventoux CCM Acquisition       10.0     9.89   
2   2020-12-23    IKT          Inhibikase Therapeutics       10.0     6.21   
3   2020-12-23    GBS                        GBS, Inc.       17.0     6.81   
4   2020-12-23   HCAR  Healthcare Services Acquisition       10.0     9.85   
..         ...    ...                              ...        ...      ...   
705 2019-01-17   RBKB                Rhinebeck Bancorp       10.0    10.49   
706 2019-01-09   BCOW        1895 Bancorp of Wisconsin       10.0    14.94   
707 2019-01-08   MDJH                        MDJM Ltd.        5.0     4.25   
708 2019-01-08    MTC                      MMTec, Inc.        4.0     2.83   
709 2019-01-04   SXTC        China SXT Pharmaceuticals        4.0     2.03   

     RETURN  
0   -0.3133  
1   -0.0110  
2   -0.3790  
3   -0.5994  
4   -0.0155  
..      ...  
705  0.0490  
706  0.4940  
707 -0.1500  
708 -0.2925  
709 -0.4925  

[710 rows x 6 columns]

In [3]:
IPO_df.dtypes

IPO_DATE     datetime64[ns]
SYMBOL               object
NAME                 object
IPO_PRICE           float64
CURRENT             float64
RETURN              float64
dtype: object

In [4]:
IPO_list = IPO_df['SYMBOL'].tolist()
IPO_list

['MRM',
 'VTAQ',
 'IKT',
 'GBS',
 'HCAR',
 'CFIV',
 'VII',
 'VHAQ',
 'ACKIT',
 'SVOK',
 'GFX',
 'MASS',
 'DUNE',
 'MTAC',
 'IIII',
 'MDWT',
 'SCOA',
 'COOL',
 'VIRI',
 'WISH',
 'UPST',
 'CCV',
 'SCPS',
 'BCAB',
 'WNW',
 'KINZ',
 'MRAC',
 'OCG',
 'BLUW',
 'ATA',
 'ABCL',
 'NBTX',
 'FDMT',
 'DWIN',
 'ROCC',
 'GLAQ',
 'MOTV',
 'EDTX',
 'SNRH',
 'GHVI',
 'TVAC',
 'CBAH',
 'VVOS',
 'CERT',
 'RMGB',
 'HYFM',
 'ABNB',
 'CTAQ',
 'PCPC',
 'NEBC',
 'DASH',
 'HMCO',
 'PUBM',
 'ALTU',
 'AI',
 'FLAC',
 'CND',
 'DDMX',
 'MUDS',
 'RAAC',
 'YQ',
 'SBTX',
 'SEER',
 'SGTX',
 'FTCV',
 'PTIC',
 'FPAC',
 'HTPA',
 'KNTE',
 'LOKB',
 'DCBO',
 'SPFR',
 'RSVA',
 'TACA',
 'CAP',
 'SGAM',
 'FRX',
 'SPRQ',
 'HFEN',
 'OZON',
 'SV',
 'VMAR',
 'GNPK',
 'VCVC',
 'TINV',
 'BREZ',
 'LSAQ',
 'ARBG',
 'MRVI',
 'CAS',
 'SHC',
 'KWAC',
 'OCA',
 'IIAC',
 'NGMS',
 'LNFA',
 'RTPZ',
 'TLS',
 'YSG',
 'OLMA',
 'JYAC',
 'CHFW',
 'PHIC',
 'DGNS',
 'PIPP',
 'OTRA',
 'ZNTE',
 'HAAC',
 'CFAC',
 'BWAC',
 'DMYI',
 'STIC',
 'NOAC',
 'TSI

In [5]:
# Check length of IPO_list
len(IPO_list)

710

#### Create a loop to perform a filter on each statement for each stock

In [6]:
# Loop through all symbols in IPO_List

FileNotFoundError_List = []

for symbol in IPO_list:
    
    
    try: 

        symbol_df = IPO_df.loc[IPO_df['SYMBOL']==symbol]
        symbol_IPO_DATE = symbol_df['IPO_DATE']

        # Define range lower bound and upper bound
        delta_days = 45

        date_lower_bound = symbol_IPO_DATE - timedelta(days=delta_days)
        date_upper_bound = symbol_IPO_DATE + timedelta(days=delta_days)

        print(f'{symbol}: {symbol_IPO_DATE}')
        print(f'{symbol} lower bound: {date_lower_bound}')
        print(f'{symbol} upper bound: {date_upper_bound}')
        print(f'----------------------------------')

        # Define dynamic path for each symbol
        symbol_cash_flow_path = f'raw_dataset/cash_flow/{symbol}_cash_flow.csv'

        # Read in each symbol's income statement
        d_parser = lambda x: pd.datetime.strptime(x,'%Y-%m-%d')
        symbol_cash_flow_df = pd.read_csv(symbol_cash_flow_path, parse_dates = ['fiscalDateEnding'], date_parser = d_parser, index_col=0)
        pd.set_option('display.max_columns',85)

        # To filter for dates within the IPO_date +/- 30 days range
        fiscalDate_filt = (symbol_cash_flow_df['fiscalDateEnding'] >= date_lower_bound.squeeze()) & (symbol_cash_flow_df['fiscalDateEnding'] <= date_upper_bound.squeeze())
        filtered_symbol_cash_flow_df = symbol_cash_flow_df.loc[fiscalDate_filt]

        # Exporting test income statement dataframe to csv file
        save_file_path = f'raw_dataset/filtered_cash_flow/{symbol}_filtered_cash_flow.csv'
        filtered_symbol_cash_flow_df.to_csv(save_file_path, index=False)
        
    except FileNotFoundError:
        print(f"FileNotFoundError occured on {symbol}")
        print(f'----------------------------------')
        FileNotFoundError_List.append(symbol)


C:\Users\User\anaconda3\envs\PythonData\lib\site-packages\ipykernel_launcher.py:28: FutureWarning: The pandas.datetime class is deprecated and will be removed from pandas in a future version. Import from datetime module instead.


MRM: 0   2020-12-29
Name: IPO_DATE, dtype: datetime64[ns]
MRM lower bound: 0   2020-11-14
Name: IPO_DATE, dtype: datetime64[ns]
MRM upper bound: 0   2021-02-12
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
VTAQ: 1   2020-12-24
Name: IPO_DATE, dtype: datetime64[ns]
VTAQ lower bound: 1   2020-11-09
Name: IPO_DATE, dtype: datetime64[ns]
VTAQ upper bound: 1   2021-02-07
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
IKT: 2   2020-12-23
Name: IPO_DATE, dtype: datetime64[ns]
IKT lower bound: 2   2020-11-08
Name: IPO_DATE, dtype: datetime64[ns]
IKT upper bound: 2   2021-02-06
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
GBS: 3   2020-12-23
Name: IPO_DATE, dtype: datetime64[ns]
GBS lower bound: 3   2020-11-08
Name: IPO_DATE, dtype: datetime64[ns]
GBS upper bound: 3   2021-02-06
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
HCAR: 4   2020-12-23
Name: IPO_DATE, dtype: datetime64[ns]
HCAR l

GHVI: 39   2020-12-11
Name: IPO_DATE, dtype: datetime64[ns]
GHVI lower bound: 39   2020-10-27
Name: IPO_DATE, dtype: datetime64[ns]
GHVI upper bound: 39   2021-01-25
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
TVAC: 40   2020-12-11
Name: IPO_DATE, dtype: datetime64[ns]
TVAC lower bound: 40   2020-10-27
Name: IPO_DATE, dtype: datetime64[ns]
TVAC upper bound: 40   2021-01-25
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
CBAH: 41   2020-12-11
Name: IPO_DATE, dtype: datetime64[ns]
CBAH lower bound: 41   2020-10-27
Name: IPO_DATE, dtype: datetime64[ns]
CBAH upper bound: 41   2021-01-25
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
VVOS: 42   2020-12-11
Name: IPO_DATE, dtype: datetime64[ns]
VVOS lower bound: 42   2020-10-27
Name: IPO_DATE, dtype: datetime64[ns]
VVOS upper bound: 42   2021-01-25
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
CERT: 43   2020-12-11
Name: IPO_DATE, dtype:

----------------------------------
SV: 80   2020-11-24
Name: IPO_DATE, dtype: datetime64[ns]
SV lower bound: 80   2020-10-10
Name: IPO_DATE, dtype: datetime64[ns]
SV upper bound: 80   2021-01-08
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
VMAR: 81   2020-11-24
Name: IPO_DATE, dtype: datetime64[ns]
VMAR lower bound: 81   2020-10-10
Name: IPO_DATE, dtype: datetime64[ns]
VMAR upper bound: 81   2021-01-08
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
GNPK: 82   2020-11-24
Name: IPO_DATE, dtype: datetime64[ns]
GNPK lower bound: 82   2020-10-10
Name: IPO_DATE, dtype: datetime64[ns]
GNPK upper bound: 82   2021-01-08
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
VCVC: 83   2020-11-24
Name: IPO_DATE, dtype: datetime64[ns]
VCVC lower bound: 83   2020-10-10
Name: IPO_DATE, dtype: datetime64[ns]
VCVC upper bound: 83   2021-01-08
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
TINV: 84   2020

CONX: 118   2020-10-30
Name: IPO_DATE, dtype: datetime64[ns]
CONX lower bound: 118   2020-09-15
Name: IPO_DATE, dtype: datetime64[ns]
CONX upper bound: 118   2020-12-14
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
SQZ: 119   2020-10-30
Name: IPO_DATE, dtype: datetime64[ns]
SQZ lower bound: 119   2020-09-15
Name: IPO_DATE, dtype: datetime64[ns]
SQZ upper bound: 119   2020-12-14
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
JUPW: 120   2020-10-30
Name: IPO_DATE, dtype: datetime64[ns]
JUPW lower bound: 120   2020-09-15
Name: IPO_DATE, dtype: datetime64[ns]
JUPW upper bound: 120   2020-12-14
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
BHSE: 121   2020-10-30
Name: IPO_DATE, dtype: datetime64[ns]
BHSE lower bound: 121   2020-09-15
Name: IPO_DATE, dtype: datetime64[ns]
BHSE upper bound: 121   2020-12-14
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
AVIR: 122   2020-10-30
Name: IPO_DA

Name: IPO_DATE, dtype: datetime64[ns]
YGMZ lower bound: 154   2020-09-06
Name: IPO_DATE, dtype: datetime64[ns]
YGMZ upper bound: 154   2020-12-05
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
LFTR: 155   2020-10-21
Name: IPO_DATE, dtype: datetime64[ns]
LFTR lower bound: 155   2020-09-06
Name: IPO_DATE, dtype: datetime64[ns]
LFTR upper bound: 155   2020-12-05
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
DCRB: 156   2020-10-20
Name: IPO_DATE, dtype: datetime64[ns]
DCRB lower bound: 156   2020-09-05
Name: IPO_DATE, dtype: datetime64[ns]
DCRB upper bound: 156   2020-12-04
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
GWAC: 157   2020-10-20
Name: IPO_DATE, dtype: datetime64[ns]
GWAC lower bound: 157   2020-09-05
Name: IPO_DATE, dtype: datetime64[ns]
GWAC upper bound: 157   2020-12-04
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
HLXA: 158   2020-10-20
Name: IPO_DATE, dtype: datetime6

Name: IPO_DATE, dtype: datetime64[ns]
APSG lower bound: 194   2020-08-18
Name: IPO_DATE, dtype: datetime64[ns]
APSG upper bound: 194   2020-11-16
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
IMNM: 195   2020-10-02
Name: IPO_DATE, dtype: datetime64[ns]
IMNM lower bound: 195   2020-08-18
Name: IPO_DATE, dtype: datetime64[ns]
IMNM upper bound: 195   2020-11-16
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
AVAN: 196   2020-10-02
Name: IPO_DATE, dtype: datetime64[ns]
AVAN lower bound: 196   2020-08-18
Name: IPO_DATE, dtype: datetime64[ns]
AVAN upper bound: 196   2020-11-16
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
ASAQ: 197   2020-10-02
Name: IPO_DATE, dtype: datetime64[ns]
ASAQ lower bound: 197   2020-08-18
Name: IPO_DATE, dtype: datetime64[ns]
ASAQ upper bound: 197   2020-11-16
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
IACA: 198   2020-10-02
Name: IPO_DATE, dtype: datetime6

Name: IPO_DATE, dtype: datetime64[ns]
GLSI lower bound: 228   2020-08-11
Name: IPO_DATE, dtype: datetime64[ns]
GLSI upper bound: 228   2020-11-09
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
PMVP: 229   2020-09-25
Name: IPO_DATE, dtype: datetime64[ns]
PMVP lower bound: 229   2020-08-11
Name: IPO_DATE, dtype: datetime64[ns]
PMVP upper bound: 229   2020-11-09
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
GRAY: 230   2020-09-25
Name: IPO_DATE, dtype: datetime64[ns]
GRAY lower bound: 230   2020-08-11
Name: IPO_DATE, dtype: datetime64[ns]
GRAY upper bound: 230   2020-11-09
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
VIAO: 231   2020-09-25
Name: IPO_DATE, dtype: datetime64[ns]
VIAO lower bound: 231   2020-08-11
Name: IPO_DATE, dtype: datetime64[ns]
VIAO upper bound: 231   2020-11-09
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
TSHA: 232   2020-09-24
Name: IPO_DATE, dtype: datetime6

Name: IPO_DATE, dtype: datetime64[ns]
OM lower bound: 264   2020-08-01
Name: IPO_DATE, dtype: datetime64[ns]
OM upper bound: 264   2020-10-30
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
SAII: 265   2020-09-15
Name: IPO_DATE, dtype: datetime64[ns]
SAII lower bound: 265   2020-08-01
Name: IPO_DATE, dtype: datetime64[ns]
SAII upper bound: 265   2020-10-30
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
SNPR: 266   2020-09-11
Name: IPO_DATE, dtype: datetime64[ns]
SNPR lower bound: 266   2020-07-28
Name: IPO_DATE, dtype: datetime64[ns]
SNPR upper bound: 266   2020-10-26
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
TWCT: 267   2020-09-11
Name: IPO_DATE, dtype: datetime64[ns]
TWCT lower bound: 267   2020-07-28
Name: IPO_DATE, dtype: datetime64[ns]
TWCT upper bound: 267   2020-10-26
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
LEAP: 268   2020-09-11
Name: IPO_DATE, dtype: datetime64[ns

Name: IPO_DATE, dtype: datetime64[ns]
FIII lower bound: 298   2020-07-04
Name: IPO_DATE, dtype: datetime64[ns]
FIII upper bound: 298   2020-10-02
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
AONE: 299   2020-08-18
Name: IPO_DATE, dtype: datetime64[ns]
AONE lower bound: 299   2020-07-04
Name: IPO_DATE, dtype: datetime64[ns]
AONE upper bound: 299   2020-10-02
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
NGA: 300   2020-08-18
Name: IPO_DATE, dtype: datetime64[ns]
NGA lower bound: 300   2020-07-04
Name: IPO_DATE, dtype: datetime64[ns]
NGA upper bound: 300   2020-10-02
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
LCAP: 301   2020-08-14
Name: IPO_DATE, dtype: datetime64[ns]
LCAP lower bound: 301   2020-06-30
Name: IPO_DATE, dtype: datetime64[ns]
LCAP upper bound: 301   2020-09-28
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
CVAC: 302   2020-08-14
Name: IPO_DATE, dtype: datetime64[n

Name: IPO_DATE, dtype: datetime64[ns]
ALVR lower bound: 338   2020-06-15
Name: IPO_DATE, dtype: datetime64[ns]
ALVR upper bound: 338   2020-09-13
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
VERX: 339   2020-07-29
Name: IPO_DATE, dtype: datetime64[ns]
VERX lower bound: 339   2020-06-14
Name: IPO_DATE, dtype: datetime64[ns]
VERX upper bound: 339   2020-09-12
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
ACEV: 340   2020-07-28
Name: IPO_DATE, dtype: datetime64[ns]
ACEV lower bound: 340   2020-06-13
Name: IPO_DATE, dtype: datetime64[ns]
ACEV upper bound: 340   2020-09-11
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
ITOS: 341   2020-07-24
Name: IPO_DATE, dtype: datetime64[ns]
ITOS lower bound: 341   2020-06-09
Name: IPO_DATE, dtype: datetime64[ns]
ITOS upper bound: 341   2020-09-07
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
INZY: 342   2020-07-24
Name: IPO_DATE, dtype: datetime6

----------------------------------
FileNotFoundError occured on TXAC
----------------------------------
RACA: 373   2020-07-08
Name: IPO_DATE, dtype: datetime64[ns]
RACA lower bound: 373   2020-05-24
Name: IPO_DATE, dtype: datetime64[ns]
RACA upper bound: 373   2020-08-22
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
LMND: 374   2020-07-02
Name: IPO_DATE, dtype: datetime64[ns]
LMND lower bound: 374   2020-05-18
Name: IPO_DATE, dtype: datetime64[ns]
LMND upper bound: 374   2020-08-16
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
CPSR: 375   2020-07-02
Name: IPO_DATE, dtype: datetime64[ns]
CPSR lower bound: 375   2020-05-18
Name: IPO_DATE, dtype: datetime64[ns]
CPSR upper bound: 375   2020-08-16
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
ACCD: 376   2020-07-02
Name: IPO_DATE, dtype: datetime64[ns]
ACCD lower bound: 376   2020-05-18
Name: IPO_DATE, dtype: datetime64[ns]
ACCD upper bound: 376   2020-08-16
N

Name: IPO_DATE, dtype: datetime64[ns]
ARYB lower bound: 411   2020-04-21
Name: IPO_DATE, dtype: datetime64[ns]
ARYB upper bound: 411   2020-07-20
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
DADA: 412   2020-06-05
Name: IPO_DATE, dtype: datetime64[ns]
DADA lower bound: 412   2020-04-21
Name: IPO_DATE, dtype: datetime64[ns]
DADA upper bound: 412   2020-07-20
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
ZI: 413   2020-06-04
Name: IPO_DATE, dtype: datetime64[ns]
ZI lower bound: 413   2020-04-20
Name: IPO_DATE, dtype: datetime64[ns]
ZI upper bound: 413   2020-07-19
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
WMG: 414   2020-06-03
Name: IPO_DATE, dtype: datetime64[ns]
WMG lower bound: 414   2020-04-19
Name: IPO_DATE, dtype: datetime64[ns]
WMG upper bound: 414   2020-07-18
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
PLRX: 415   2020-06-03
Name: IPO_DATE, dtype: datetime64[ns]
PLR

PASG: 448   2020-02-28
Name: IPO_DATE, dtype: datetime64[ns]
PASG lower bound: 448   2020-01-14
Name: IPO_DATE, dtype: datetime64[ns]
PASG upper bound: 448   2020-04-13
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
ZCMD: 449   2020-02-24
Name: IPO_DATE, dtype: datetime64[ns]
ZCMD lower bound: 449   2020-01-10
Name: IPO_DATE, dtype: datetime64[ns]
ZCMD upper bound: 449   2020-04-09
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
DMYT: 450   2020-02-21
Name: IPO_DATE, dtype: datetime64[ns]
DMYT lower bound: 450   2020-01-07
Name: IPO_DATE, dtype: datetime64[ns]
DMYT upper bound: 450   2020-04-06
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
FileNotFoundError occured on DMYT
----------------------------------
ESSC: 451   2020-02-20
Name: IPO_DATE, dtype: datetime64[ns]
ESSC lower bound: 451   2020-01-06
Name: IPO_DATE, dtype: datetime64[ns]
ESSC upper bound: 451   2020-04-05
Name: IPO_DATE, dtype: datetime64[ns

Name: IPO_DATE, dtype: datetime64[ns]
OCFT lower bound: 486   2019-10-29
Name: IPO_DATE, dtype: datetime64[ns]
OCFT upper bound: 486   2020-01-27
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
BILL: 487   2019-12-12
Name: IPO_DATE, dtype: datetime64[ns]
BILL lower bound: 487   2019-10-28
Name: IPO_DATE, dtype: datetime64[ns]
BILL upper bound: 487   2020-01-26
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
EH: 488   2019-12-12
Name: IPO_DATE, dtype: datetime64[ns]
EH lower bound: 488   2019-10-28
Name: IPO_DATE, dtype: datetime64[ns]
EH upper bound: 488   2020-01-26
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
XP: 489   2019-12-11
Name: IPO_DATE, dtype: datetime64[ns]
XP lower bound: 489   2019-10-27
Name: IPO_DATE, dtype: datetime64[ns]
XP upper bound: 489   2020-01-25
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
LIVK: 490   2019-12-11
Name: IPO_DATE, dtype: datetime64[ns]
LIVK l

Name: IPO_DATE, dtype: datetime64[ns]
AIH lower bound: 524   2019-09-10
Name: IPO_DATE, dtype: datetime64[ns]
AIH upper bound: 524   2019-12-09
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
CABA: 525   2019-10-25
Name: IPO_DATE, dtype: datetime64[ns]
CABA lower bound: 525   2019-09-10
Name: IPO_DATE, dtype: datetime64[ns]
CABA upper bound: 525   2019-12-09
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
HAPP: 526   2019-10-25
Name: IPO_DATE, dtype: datetime64[ns]
HAPP lower bound: 526   2019-09-10
Name: IPO_DATE, dtype: datetime64[ns]
HAPP upper bound: 526   2019-12-09
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
BRP: 527   2019-10-24
Name: IPO_DATE, dtype: datetime64[ns]
BRP lower bound: 527   2019-09-09
Name: IPO_DATE, dtype: datetime64[ns]
BRP upper bound: 527   2019-12-08
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
GLEO: 528   2019-10-18
Name: IPO_DATE, dtype: datetime64[ns]

SNDL: 564   2019-08-01
Name: IPO_DATE, dtype: datetime64[ns]
SNDL lower bound: 564   2019-06-17
Name: IPO_DATE, dtype: datetime64[ns]
SNDL upper bound: 564   2019-09-15
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
DT: 565   2019-08-01
Name: IPO_DATE, dtype: datetime64[ns]
DT lower bound: 565   2019-06-17
Name: IPO_DATE, dtype: datetime64[ns]
DT upper bound: 565   2019-09-15
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
NFIN: 566   2019-07-31
Name: IPO_DATE, dtype: datetime64[ns]
NFIN lower bound: 566   2019-06-16
Name: IPO_DATE, dtype: datetime64[ns]
NFIN upper bound: 566   2019-09-14
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
FileNotFoundError occured on NFIN
----------------------------------
BORR: 567   2019-07-31
Name: IPO_DATE, dtype: datetime64[ns]
BORR lower bound: 567   2019-06-16
Name: IPO_DATE, dtype: datetime64[ns]
BORR upper bound: 567   2019-09-14
Name: IPO_DATE, dtype: datetime64[ns]
----

AKRO: 603   2019-06-20
Name: IPO_DATE, dtype: datetime64[ns]
AKRO lower bound: 603   2019-05-06
Name: IPO_DATE, dtype: datetime64[ns]
AKRO upper bound: 603   2019-08-04
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
PSNL: 604   2019-06-20
Name: IPO_DATE, dtype: datetime64[ns]
PSNL lower bound: 604   2019-05-06
Name: IPO_DATE, dtype: datetime64[ns]
PSNL upper bound: 604   2019-08-04
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
GO: 605   2019-06-20
Name: IPO_DATE, dtype: datetime64[ns]
GO lower bound: 605   2019-05-06
Name: IPO_DATE, dtype: datetime64[ns]
GO upper bound: 605   2019-08-04
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
BCEL: 606   2019-06-20
Name: IPO_DATE, dtype: datetime64[ns]
BCEL lower bound: 606   2019-05-06
Name: IPO_DATE, dtype: datetime64[ns]
BCEL upper bound: 606   2019-08-04
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
WORK: 607   2019-06-20
Name: IPO_DATE,

YJ: 643   2019-05-03
Name: IPO_DATE, dtype: datetime64[ns]
YJ lower bound: 643   2019-03-19
Name: IPO_DATE, dtype: datetime64[ns]
YJ upper bound: 643   2019-06-17
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
RRBI: 644   2019-05-03
Name: IPO_DATE, dtype: datetime64[ns]
RRBI lower bound: 644   2019-03-19
Name: IPO_DATE, dtype: datetime64[ns]
RRBI upper bound: 644   2019-06-17
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
ATIF: 645   2019-05-03
Name: IPO_DATE, dtype: datetime64[ns]
ATIF lower bound: 645   2019-03-19
Name: IPO_DATE, dtype: datetime64[ns]
ATIF upper bound: 645   2019-06-17
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
BYND: 646   2019-05-02
Name: IPO_DATE, dtype: datetime64[ns]
BYND lower bound: 646   2019-03-18
Name: IPO_DATE, dtype: datetime64[ns]
BYND upper bound: 646   2019-06-16
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
SY: 647   2019-05-02
Name: IPO_DATE, d

THCB: 681   2019-03-05
Name: IPO_DATE, dtype: datetime64[ns]
THCB lower bound: 681   2019-01-19
Name: IPO_DATE, dtype: datetime64[ns]
THCB upper bound: 681   2019-04-19
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
HCAC: 682   2019-03-01
Name: IPO_DATE, dtype: datetime64[ns]
HCAC lower bound: 682   2019-01-15
Name: IPO_DATE, dtype: datetime64[ns]
HCAC upper bound: 682   2019-04-15
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
FileNotFoundError occured on HCAC
----------------------------------
KLDO: 683   2019-02-28
Name: IPO_DATE, dtype: datetime64[ns]
KLDO lower bound: 683   2019-01-14
Name: IPO_DATE, dtype: datetime64[ns]
KLDO upper bound: 683   2019-04-14
Name: IPO_DATE, dtype: datetime64[ns]
----------------------------------
SHLL: 684   2019-02-28
Name: IPO_DATE, dtype: datetime64[ns]
SHLL lower bound: 684   2019-01-14
Name: IPO_DATE, dtype: datetime64[ns]
SHLL upper bound: 684   2019-04-14
Name: IPO_DATE, dtype: datetime64[ns

In [7]:
# Print length of FileNotFoundError List
len(FileNotFoundError_List)

48

This is expected. During the process of retrieving the quarterly cash flow on Alpha Vantage for each symbol, 48 our of 710 symbols on the IPO_List had occured a KeyError where quarterlyReports were not found in the API response.